Casting bug when using interface

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: windows, Microsoft Windows 11 Pro v22000 64-bit
Processor: 8 cpus, 64bit, little endian, Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz  

CC version: Error: exec failed (CreateProcess) with code 2: The system cannot find the file specified.

 cmd: cc --version

getwd: C:\Users\kylep\Desktop\learning-v
vmodules: C:\Users\kylep\.vmodules
vroot: C:\v
vexe: C:\v\v.exe
vexe mtime: 2022-03-17 16:14:49
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 7f62346.136f39a

Git version: git version 2.35.0.windows.1
Git vroot status: weekly.2022.10-45-g136f39a2-dirty
.git/config present: true
thirdparty/tcc status: thirdparty-windows-amd64 9de1a49e

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg any_interface.v`
```v
interface Any{}

fn print_out(x Any) {
	if x is string {
		println(x)
	}
}

fn main(){
	print_out('12345')
}
```

**What did you expect to see?**

Print out string passed to function. From my understanding of the docs this should be possible...?


**What did you see instead?**
```
==================
C:/Users/kylep/AppData/Local/Temp/v_0/any_interface.12993884173728592741.tmp.c:6879: warning: cast between pointer and integer of different size
C:/Users/kylep/AppData/Local/Temp/v_0/any_interface.12993884173728592741.tmp.c:11465: error: '_main__Any__v_string_index' undeclared
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/gen/c/infix_expr.v
+++ b/vlib/v/gen/c/infix_expr.v
@@ -507,7 +507,7 @@ fn (mut g Gen) infix_expr_is_op(node ast.InfixExpr) {
 			else { ast.Type(0) }
 		}
 		sub_sym := g.table.sym(sub_type)
-		g.write('_${c_name(sym.name)}_${c_name(sub_sym.name)}_index')
+		g.write('_${sym.cname}_${sub_sym.cname}_index')
 		return
 	} else if sym.kind == .sum_type {
 		g.write('_typ $cmp_op ')

--- a/vlib/v/tests/empty_interface_test.v
+++ b/vlib/v/tests/empty_interface_test.v
@@ -1,0 +1,14 @@
+interface Any {}
+
+fn print_out(x Any) string {
+	if x is string {
+		println(x)
+		return '$x'
+	}
+	return ''
+}
+
+fn test_empty_interface() {
+	ret := print_out('12345')
+	assert ret == '&12345'
+}

